### ISSUE TITLE
Techniques for Optimizing hyperparameters in the verl Framework for Improved Training Efficiency

### ISSUE THREAD

**Issue by LLouice** [agentica-project/deepscaler#4]
Hi, amazing work! Thanks for sharing. I am currently exploring ways to enhance training efficiency in verl. Given that reinforcement learning can be quite slow and requires extensive training time, I was wondering if you have any tips or best practices for tuning the hyperparameters in verl, such as batch size, tp, rollout.n, etc., to achieve optimal performance. Any insights or suggestions would be greatly appreciated.

Thank you!

**Comment by michaelzhiluo** [comment]
Great question! 

Reinforcement learning for LLM reasoning is inherently very slow, for 24k runs it was around 600-800(s) per iteration.

Some tips:
- RL is bounded by sampling time: Argmax Data parallelism (# of engines) and argmin tensor paralleism
- Training parameters don't really affect time. For performance, always set ppo_epochs=1, as ppo_epochs>1 leads to overfitting.
- Recommend having 8,16, or even 32 samples per problem. Batch size should be at least 128. PPO/GRPO policy gradient is very high variance and you can see our validation curve vacillating quite a lot.
